setup.py

```diff
@@ -1,10 +1,29 @@
-import os
 from setuptools import setup
-from setuptools.config import read_configuration
+import ast
+try:
+    from pathlib import Path
+except:
+    from pathlib2 import Path
 
-this_dir = os.path.dirname(__file__)
-cfg = read_configuration(os.path.join(this_dir, 'setup.cfg'))
-cfg["options"].update(cfg["metadata"])
-cfg = cfg["options"]
 
-setup(**cfg)
+def extract_global_vars(file):
+    t = file.read_text(encoding="utf-8")
+    m = ast.parse(t)
+    res = {}
+    for s in m.body:
+        if isinstance(s, ast.Assign):
+            vo = s.value
+            v = None
+            if isinstance(vo, ast.Str):
+                v = vo.s
+            elif isinstance(vo, ast.Num):
+                v = vo.n
+            
+            if v is not None:
+                for t in s.targets:
+                    if isinstance(t, ast.Name):
+                        res[t.id] = v
```

Comment:
This isn't how assignment to multiple variables works. `x, y = 1, 2` sets `x = 1` and `y = 2`, but your code interprets it as `x = (1, 2)`, `y = (1, 2)`.

For simplicity I would ignore complex assignments with multiple targets, unpacking, etc. - realistically, `__version__` will only be set with a simple single-target assignment.

Reply:
you are wrong. https://ideone.com/xashTQ

Reply:
Ah, you're right, my bad. In that case it actually makes sense to handle it this way (although IMHO it's still kind of unlikely that `__version__` would be set in a multi-target assignment).